VERSIONING.md

```diff
@@ -0,0 +1,119 @@
+# Versioning
+
+This document aims to provide an overview of how VAST is versioned, and what
+significance a version bump carries.
+
+Run the `vast version` command to get VAST's current version, and the versions
+of its major dependencies that it currently uses:
+
+```json
+{
+  "VAST": "v1.0.4-61-geed8cb41a3",
+  "VAST Build Tree Hash": "0eb048d0eeda5be7081f0c3a659ff17f",
+  "CAF": "0.17.6",
+  "Apache Arrow": "6.0.1",
+  "jemalloc": "5.2.1",
+  "plugins": {
+    "compaction": "v1.2.0-61-g9131346958",
+    "matcher": "v2.1.4-23-gfb69e84ade",
+  }
+}
+```
+
+## Version Components
+
+VAST's version number consists of three mandatory and two optional parts:
+`v<major>.<minor>.<patch>[-rc<candidate>][-<tag>]`. Every release on the default
```

Comment:
```suggestion
`v<major>.<minor>.<patch>[-rc<candidate>][-<tag>]`. Every release on the default
```

Have we ever discussed the `v` prefix? We currently don't use this, and I like it that way because it makes for easier parsing.

Reply:
It's a sorting thing: The CalVer releases would always show up as later releases when sorting reverse-lexicographically until we hit major version 2022. With the prefix, reverse-lexicographical sorting equals sorting by newest release date.

On a side note, GitHub recommends using a `v` prefix for release tags. They refused to let you create a tag that wasn't prefixed with `v` from the releases page up until a few months ago when they redesigned it.

Reply:
> It's a sorting thing

Okay, I can see that.

> On a side note, GitHub recommends using a `v` prefix for release tags.

Interesting, can you provide a source?

Reply:
This is on the page for creating a new release:

<img width="347" alt="image" src="https://user-images.githubusercontent.com/4488655/149925985-143f9d08-4a4f-454e-a678-9a1e11c1ce4c.png">